src/sentry_webhook/post_sentry_event_to_salesforce.ts

```diff
@@ -0,0 +1,162 @@
+  os: string;
+  version: string;
+  type: string;
+}
+
+// Defines the Salesforce form values.
+interface SalesforceFormValues {
+  orgId: string;
+  recordType: string;
+}
+
+const SALESFORCE_DEV_HOST = 'google-jigsaw--uat.my.salesforce.com';
+const SALESFORCE_PROD_HOST = 'webto.salesforce.com';
+const SALESFORCE_PATH = '/servlet/servlet.WebToCase';
+const SALESFORCE_FORM_FIELDS_DEV: SalesforceFormFields = {
+  orgId: 'orgid',
+  recordType: 'recordType',
+  email: 'email',
+  description: 'description',
+  category: '00N3F000002Rqho',
+  cloudProvider: '00N3F000002Rqhs',
+  sentryEventId: '00N3F000002Rqhq',
+  os: '00N3F000002cLcN',
+  version: '00N3F000002cLcI',
+  type: 'type'
+};
+const SALESFORCE_FORM_FIELDS_PROD: SalesforceFormFields = {
+  orgId: 'orgid',
+  recordType: 'recordType',
+  email: 'email',
+  description: 'description',
+  category: '00N0b00000BqOA2',
+  cloudProvider: '00N0b00000BqOA7',
+  sentryEventId: '00N0b00000BqOA4',
+  os: '00N0b00000BqOfW',
+  version: '00N0b00000BqOfR',
+  type: 'type'
+};
+const SALESFORCE_FORM_VALUES_DEV: SalesforceFormValues = {
+  orgId: '00D3F000000DDDH',
+  recordType: '0123F000000MWTS',
+};
+const SALESFORCE_FORM_VALUES_PROD: SalesforceFormValues = {
+  orgId: '00D0b000000BrsN',
+  recordType: '0120b0000006e8i',
+};
+
+// Posts a Sentry event to Salesforce using predefined form data. Only posts if the event contains
+// an email address.
+export function postSentryEventToSalesforce(
+    event: sentry.SentryEvent, project: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if (!event.user || !event.user.email) {
+      reject(new Error('Not posting event without email'));
+      return;
+    }
+    // Sentry development projects are marked with 'dev', i.e. outline-client-dev.
+    const isProd = project.indexOf('-dev') === -1;
+    const salesforceHost = isProd ? SALESFORCE_PROD_HOST : SALESFORCE_DEV_HOST;
+    const formFields = isProd ? SALESFORCE_FORM_FIELDS_PROD : SALESFORCE_FORM_FIELDS_DEV;
+    const formValues = isProd ? SALESFORCE_FORM_VALUES_PROD : SALESFORCE_FORM_VALUES_DEV;
+    const isClient = project.indexOf('client') !== -1;
+    const formData =
+        getSalesforceFormData(formFields, formValues, event, event.user.email, isClient);
+    const req = https.request(
+        {
+          host: salesforceHost,
+          path: SALESFORCE_PATH,
+          protocol: 'https:',
+          method: 'post',
+          headers: {
+            // The production server will reject requests that do not specify this content type.
+            'Content-Type': 'application/x-www-form-urlencoded'
+          }
+        },
+        (res) => {
+          if (res.statusCode === 200) {
+            resolve();
+          } else {
+            reject(new Error(`Failed to post form data, response status: ${res.statusCode}`));
+          }
+        });
+    req.on('error', (err) => {
+      reject(new Error(
+          `Failed to submit form: ${err}`,
```

Comment:
the trailing comma is messing up the formatting here

Reply:
Done in 4d60e67.